pkg/istio-agent/xds_proxy.go

```diff
@@ -224,6 +231,27 @@ func (p *XdsProxy) StreamAggregatedResources(downstream discovery.AggregatedDisc
 				proxyLog.Errorf("upstream send error for type url %s: %v", req.TypeUrl, err)
 				return err
 			}
+		case healthEvent, ok := <-healthEventsChan:
+			if ok {
```

Comment:
nit: prefer to break early if !ok rather than keep everything in branch, as a code style. See https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement for details